alert-common/src/test/java/com/synopsys/integration/alert/common/persistence/model/ConfigurationModelTest.java

```diff
@@ -0,0 +1,80 @@
+package com.synopsys.integration.alert.common.persistence.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.persistence.model.mutable.ConfigurationModelMutable;
+
+public class ConfigurationModelTest {
+    private final Long descriptorId = 1L;
+    private final Long configurationId = 2L;
+    private final String createdAt = "createdAt-test";
+    private final String lastUpdated = "lastUpdated-test";
+    private final ConfigContextEnum configContextEnum = ConfigContextEnum.GLOBAL;
+    private final String fieldKey = "fieldKey";
+
+    @Test
+    public void getDescriptorIdTest() {
+        ConfigurationModel configurationModel = createConfigurationModel();
+        assertEquals(descriptorId, configurationModel.getDescriptorId());
+    }
+
+    @Test
+    public void getConfigurationIdTest() {
+        ConfigurationModel configurationModel = createConfigurationModel();
+        assertEquals(configurationId, configurationModel.getConfigurationId());
+    }
+
+    @Test
+    public void getCreatedAtTest() {
+        ConfigurationModel configurationModel = createConfigurationModel();
+        assertEquals(createdAt, configurationModel.getCreatedAt());
+    }
+
+    @Test
+    public void getLastUpdatedTest() {
+        ConfigurationModel configurationModel = createConfigurationModel();
+        assertEquals(lastUpdated, configurationModel.getLastUpdated());
+    }
+
+    @Test
+    public void getDescriptorContextTest() {
+        ConfigurationModel configurationModel = createConfigurationModel();
+        assertEquals(configContextEnum, configurationModel.getDescriptorContext());
+    }
+
+    @Test
+    public void getFieldTest() {
+        ConfigurationModel configurationModel = createConfigurationModel();
+        assertFalse(configurationModel.getField(fieldKey).isPresent());
+    }
+
+    @Test
+    public void getCopyOfFieldListTest() {
+        ConfigurationModel configurationModel = createConfigurationModel();
+        assertTrue(configurationModel.getCopyOfFieldList().isEmpty());
+    }
+
+    @Test
+    public void getCopyOfKeyToFieldMapTest() {
+        ConfigurationModel configurationModel = createConfigurationModel();
+        assertTrue(configurationModel.getCopyOfKeyToFieldMap().isEmpty());
+    }
+
+    @Test
+    public void createMutableCopyTest() {
+        ConfigurationModel configurationModel = createConfigurationModel();
```

Comment:
Let's add a field or two to `configurationModel` to make sure that the fields are copied.

Reply:
If all of its constructors create the `Map` maybe we should consider passing it as a parameter in one of them.

Reply:
configurationModel does not have access to the put method so it cannot add a configuredField. I would be able to add a field on the configurationModelMutable but that test would be covered in ConfigurationModelMutableTest.

Reply:
Ok, that makes me think we will need to update the constructor(s) in `ConfigurationModel` then.